std/range/package.d

```diff
@@ -6590,11 +6595,53 @@ if (isForwardRange!RangeOfRanges &&
     }
 
     auto opSlice() { return this; }
+    
+    static if (hasSlicing!(ElementType!RangeOfRanges))
+    {
+        auto opSlice(int start, int end)
+        {
+            return _input.map!(a => a[start..end]).array.transposed;
+        }
+    }
 
 private:
     RangeOfRanges _input;
 }
 
+struct TransposedIndex(RangeOfRanges)
```

Comment:
Since we're requiring random access on subranges in order for `opIndex` to be possible, what about doing something similar to what `.front` does? I.e., something like:

````
// Warning: untested code
return _input.save
             .filter!(a => !a.empty)
             .map!(a => a[i]);
````